src/Components/Web.Extensions/src/InputFile/InputFile.cs

```diff
@@ -0,0 +1,108 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Runtime.Versioning;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Components.Rendering;
+using Microsoft.JSInterop;
+
+namespace Microsoft.AspNetCore.Components.Web.Extensions
+{
+    /// <summary>
+    /// A component that wraps the HTML file input element and exposes a <see cref="Stream"/> for each file's contents.
+    /// </summary>
+    public class InputFile : ComponentBase, IInputFileJsCallbacks, IDisposable
+    {
+        private ElementReference _inputFileElement;
+
+        private IJSUnmarshalledRuntime? _jsUnmarshalledRuntime;
+
+        private InputFileJsCallbacksRelay? _jsCallbacksRelay;
+
+        [Inject]
+        private IJSRuntime JSRuntime { get; set; } = default!;
+
+        /// <summary>
+        /// Gets or sets the event callback that will be invoked when the collection of selected files changes.
+        /// </summary>
+        [Parameter]
+        public EventCallback<InputFileChangeEventArgs> OnChange { get; set; }
+
+        /// <summary>
+        /// Gets or sets the maximum chunk size for file data sent over a SignalR circuit.
+        /// This only has an effect when using Blazor Server.
+        /// </summary>
+        [Parameter]
+        [UnsupportedOSPlatform("browser")]
+        public int MaxSignalRChunkSize { get; set; } = 20 * 1024; // SignalR limit is 32K.
+
+        /// <summary>
+        /// Gets or sets the maximum internal buffer size for unread data sent over a SignalR circuit.
+        /// This only has an effect when using Blazor Server.
+        /// </summary>
+        [Parameter]
+        [UnsupportedOSPlatform("browser")]
+        public int MaxUnreadMemoryBufferSize { get; set; } = 1024 * 1024;
```

Comment:
I think my comment was missed. Should these values be settable through options so that they don't have to be specified locally if you want to change the value globally? Specially the MaxSignalRChunkSize, that is an implementation detail leaking into the component.

Reply:
I had a thought here:

```
[Inject] private IOptions<RemoteBrowserFileStreamOptions> _options;
```

```
public class RemoteBrowserFileStreamOptions
{
  public int ChunkSize { get; set; } = 20 * 1024
  public int MaxBufferSize { get; set; } = 1024 * 1024;
}
```

Users who want to customize this do

```
services.Configure<RemoteBrowserFileOptions>(o => {
  o => o.ChunkSize = 30 * 1024;
  o => o.MaxBufferSize = 2048 * 2048;
});
```

Other users don't have to worry about it.